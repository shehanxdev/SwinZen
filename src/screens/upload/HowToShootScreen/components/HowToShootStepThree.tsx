import { View } from 'react-native';
import FastImage from 'react-native-fast-image';

import { Text } from '@sz/components';
import { tw } from '@sz/config';
import { Color, FaceOnGIF, LandscapeGIF, TextAlignment, TextVariant, TripodHeightGIF } from '@sz/constants';

export function HowToShootStepThree() {
  return (
    <View>
      <Text color={Color.Primary.Sz200} variant={TextVariant.Body1SemiBold} textAlign={TextAlignment.Auto}>
        {`Face-On/Tripod Instructions\n`}
      </Text>
      <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
        {`Take 3 paces/9 ft from the ball, directly in front of the golfer.\n`}
      </Text>
      <FastImage
        style={tw`w-89.5 h-51.5 border-4 border-Neutral-White`}
        source={{ cache: FastImage.cacheControl.immutable, priority: FastImage.priority.high, uri: FaceOnGIF }}
      />
      <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
        {`\nSet your tripod 4-5 feet high.\n`}
      </Text>
      <FastImage
        style={tw`w-89.5 h-51.5 border-4 border-Neutral-White`}
        source={{ cache: FastImage.cacheControl.immutable, priority: FastImage.priority.high, uri: TripodHeightGIF }}
      />
      <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
        {`\nPosition your phone in landscape mode.\n`}
      </Text>
      <FastImage
        style={tw`w-89.5 h-51.5 border-4 border-Neutral-White`}
        source={{ cache: FastImage.cacheControl.immutable, priority: FastImage.priority.high, uri: LandscapeGIF }}
      />
    </View>
  );
}
