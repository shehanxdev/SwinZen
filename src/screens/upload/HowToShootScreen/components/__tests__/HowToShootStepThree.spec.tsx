import { render } from '@testing-library/react-native';
import * as React from 'react';

import { HowToShootStepThree } from '../HowToShootStepThree';

describe('How to shoot step three', () => {
  const getRenderedScreen = () => render(<HowToShootStepThree />);

  it(`Should render How to shoot step three correctly`, () => {
    const renderer = getRenderedScreen();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });
});
