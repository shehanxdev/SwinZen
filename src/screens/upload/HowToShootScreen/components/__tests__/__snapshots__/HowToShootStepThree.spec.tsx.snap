// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`How to shoot step three Should render How to shoot step three correctly 1`] = `
<View>
  <Text
    style={
      [
        {
          "color": "#C0DBBF",
          "lineHeight": 28,
        },
        {
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "auto",
        },
      ]
    }
  >
    Face-On/Tripod Instructions

  </Text>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "lineHeight": 28,
        },
        {
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 18,
          "fontWeight": "normal",
          "textAlign": "auto",
        },
      ]
    }
  >
    Take 3 paces/9 ft from the ball, directly in front of the golfer.

  </Text>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderColor": "#FFFFFF",
          "borderWidth": 4,
          "height": 206,
          "width": "100%",
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      source={
        {
          "cache": "immutable",
          "priority": "high",
          "uri": "https://thumbnails.dev.swingzen.com/gifs/faceon.gif",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "lineHeight": 28,
        },
        {
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 18,
          "fontWeight": "normal",
          "textAlign": "auto",
        },
      ]
    }
  >
    
Set your tripod 4-5 feet high.

  </Text>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderColor": "#FFFFFF",
          "borderWidth": 4,
          "height": 206,
          "width": "100%",
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      source={
        {
          "cache": "immutable",
          "priority": "high",
          "uri": "https://thumbnails.dev.swingzen.com/gifs/tripodheight.gif",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "lineHeight": 28,
        },
        {
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 18,
          "fontWeight": "normal",
          "textAlign": "auto",
        },
      ]
    }
  >
    
Position your phone in landscape mode.

  </Text>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderColor": "#FFFFFF",
          "borderWidth": 4,
          "height": 206,
          "width": "100%",
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      source={
        {
          "cache": "immutable",
          "priority": "high",
          "uri": "https://thumbnails.dev.swingzen.com/gifs/Landscape.gif",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
